npm/pkg/dataplane/dataplane.go

```diff
@@ -284,15 +284,36 @@ func (dp *DataPlane) RemovePolicy(policyKey string) error {
 	// because policy Manager will remove from policy from cache
 	// keep a local copy to remove references for ipsets
 	policy, ok := dp.policyMgr.GetPolicy(policyKey)
+	endpoints := make(map[string]string, len(policy.PodEndpoints))
+
+	for podIP, endpointID := range policy.PodEndpoints {
+		endpoints[podIP] = endpointID
+	}
+
 	if !ok {
 		klog.Infof("[DataPlane] Policy %s is not found. Might been deleted already", policyKey)
 		return nil
 	}
+	dp.endpointCache.Lock()
```

Comment:
Why cant we take endpointCache after the RemovePolicy ? if a new ep is created or existing one is deleted we will be fine right ?

Reply:
@huntergregory

Reply:
discussed offline:
The goal of this was to not block NetPol controller on making SetPolicy SysCalls while UpdatePod() is running. 

New Proposal:
1. Put endpoint lock below removePolicy()
2. Address the blocking in another PR that makes an endpoint-level lock (instead of locking the whole cache)

Reply:
updated 1